src/composables/maxAmount.ts

```diff
@@ -72,12 +79,12 @@ export function useMaxAmount({ store, formModel }: MaxAmountOptions) {
 
       if (val.selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
         if (
-          !tokenInstance.value
-          || tokenInstance.value.deployInfo.address !== val.selectedAsset.contractId
+          !tokenInstance
+          || tokenInstance.$options.address !== val.selectedAsset.contractId
         ) {
-          tokenInstance.value = await sdk.getContractInstance({
+          tokenInstance = await sdk.initializeContract({
             aci: FungibleTokenFullInterfaceACI,
-            contractAddress: val.selectedAsset.contractId,
+            address: val.selectedAsset.contractId as any,
```

Comment:
Couldn't we use here `Encoding.ContractAddress` as type?

Reply:
Either way it would require a proper refactoring to make `IToken` type contain the correct type for the `contractId`.